src/engine/sparqlExpressions/PrefilterExpressionIndex.cpp

```diff
@@ -489,33 +488,21 @@ template class LogicalExpression<LogicalOperator::OR>;
 namespace detail {
 
 //______________________________________________________________________________
+// Returns the corresponding mirrored `RelationalExpression<mirrored
+// comparison>` for the given `CompOp comparison` template argument. For
+// example, the mirroring procedure will transform the relational expression
+// `referenceValue > ?var` into `?var < referenceValue`.
 template <CompOp comparison>
 static std::unique_ptr<PrefilterExpression> makeMirroredExpression(
     const IdOrLocalVocabEntry& referenceValue) {
   using enum CompOp;
-  switch (comparison) {
-    case LT:
-      // Id < ?var -> ?var > Id
-      return std::make_unique<GreaterThanExpression>(referenceValue);
-    case LE:
-      // Id <= ?var -> ?var >= Id
-      return std::make_unique<GreaterEqualExpression>(referenceValue);
-    case GE:
-      // Id >= ?var -> ?var <= Id
-      return std::make_unique<LessEqualExpression>(referenceValue);
-    case GT:
-      // Id > ?var -> ?var < Id
-      return std::make_unique<LessThanExpression>(referenceValue);
-    case EQ:
-    case NE:
-      // EQ(==) or NE(!=)
-      // Given that these two relations are symmetric w.r.t. ?var and Id,
-      // no swap regarding the relational operator is necessary.
-      return std::make_unique<RelationalExpression<comparison>>(referenceValue);
-    default:
-      // Unchecked / new valueIdComparators::Comparison case.
-      AD_FAIL();
-  }
+  using P = std::pair<CompOp, CompOp>;
+  using namespace ad_utility;
+  constexpr static std::array mirrorPairs{P{LT, GT}, P{LE, GE}, P{GE, LE},
+                                          P{GT, LT}, P{EQ, EQ}, P{NE, NE}};
+  constexpr ConstexprMap<CompOp, CompOp, 6> mirrorMap(mirrorPairs);
```

Comment:
```suggestion
  constexpr ConstexprMap<CompOp, CompOp, 6> mirrorMap({{LT, GT}, {LE, GE}, ...});
```
(should work , no need for the `using P`, `using namespace` and explicit `array mirrorPairs`.

Reply:
W.r.t. the `std::pair<CompOp, CompOp>` values it seems like that they can't be automatically constructed within an initializer list, still had to use `P = std::pair<CompOp, CompOp>`.